Enhancement/2256 Separate Legacy Components

benbowler opened the pull request:
## Summary

I moved legacy setup components into a new folder called legacy-setup and refactored `SetupWrapper` to become `ModuleSetup`.
Addresses issue #2256

## Relevant technical choices

This was a simple reorganization, no technical comments.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

felixarntz commented on assets/js/components/legacy-setup/wizard-step-search-console-property.js:
```diff
@@ -29,7 +29,7 @@ import { Component } from '@wordpress/element';
  * Internal dependencies
  */
-import SearchConsole from '../../components/setup/search-console';
+import SearchConsole from '../../components/legacy-setup/search-console';
```
This could now import from `./search-console`. A similar simplification can be made to a few other imports that import from the same `legacy-setup` directory which the importing file is already in.

benbowler replied:
👌

felixarntz commented on assets/js/components/legacy-setup/wizard-step-verification.js:
```diff
@@ -29,7 +29,7 @@ import { Component } from '@wordpress/element';
  * Internal dependencies
  */
-import SiteVerification from '../../components/setup/site-verification';
+import SiteVerification from '../../components/legacy-setup/site-verification';
```
See above:
```suggestion
import SiteVerification from './site-verification';
```

felixarntz commented on assets/js/components/legacy-setup/wizard-steps.js:
```diff
@@ -55,9 +55,9 @@ const STEPS = {
-import WizardStepVerification from '../../components/setup-wizard/wizard-step-verification';
-import WizardStepSearchConsoleProperty from '../../components/setup-wizard/wizard-step-search-console-property';
-import WizardStepCompleteSetup from '../../components/setup-wizard/wizard-step-complete-setup';
+import WizardStepAuthentication from '../../components/legacy-setup/wizard-step-authentication';
+import WizardStepVerification from '../../components/legacy-setup/wizard-step-verification';
+import WizardStepSearchConsoleProperty from '../../components/legacy-setup/wizard-step-search-console-property';
+import WizardStepCompleteSetup from '../../components/legacy-setup/wizard-step-complete-setup';
```
See above.

felixarntz commented on assets/js/components/setup.js:
```diff
@@ -35,8 +35,8 @@ import Button from '../components/button';
-import STEPS from '../components/setup-wizard/wizard-steps';
-import WizardProgressStep from '../components/setup-wizard/wizard-progress-step';
+import STEPS from '../components/legacy-setup/wizard-steps';
+import WizardProgressStep from '../components/legacy-setup/wizard-progress-step';
```
The imports here can be simplified to be relative to the `assets/js/components` directory, similar to the previous comment. Would be good to also fix that for the untouched imports above where this is also unnecessary.

felixarntz reviewed (changes requested):
LGTM mostly, just some nit-pick feedback that some import references can be simplified.

Also, this will require a few updates now that #2259 has been merged, where code was changed that references some of these components.

benbowler commented:
I've merged in the changes from `develop` and fixed conflicts with the ModuleSetup component plus I've simplified imports across multiple setup components.

tofumatt reviewed (approved):
Good call on the import simplification 👍🏻

aaemnnosttv commented:
@felixarntz should we not also move `assets/js/components/setup.js` to `assets/js/components/legacy-setup/setup.js`?
(perhaps as `SetupUsingGCP.js`)

It is only referenced by legacy components and is only concerned with the legacy wizard-based setup:
* https://github.com/google/site-kit-wp/blob/60be3ea14beb5c4bfbd61359d0808efc10fa8137/assets/js/components/dashboard-splash/LegacyDashboardSplashApp.js#L31
* https://github.com/google/site-kit-wp/blob/0021b342c1869dcb8e49f771bee99ad32bd1d871/stories/setup.stories.js#L9

felixarntz commented:
@aaemnnosttv 
> @felixarntz should we not also move `assets/js/components/setup.js` to `assets/js/components/legacy-setup/setup.js`?
> (perhaps as `SetupUsingGCP.js`)
> 
> It is only referenced by legacy components and is only concerned with the legacy wizard-based setup

Good catch, let's do that! @benbowler If you could take care of that, that'd be great, but otherwise no big deal, this little change I could also add to the PR before merge.

felixarntz reviewed (dismissed):
Great!
